#! /bin/sh /usr/share/dpatch/dpatch-run
## 19-gcc43.dpatch by Thomas Girard <thomas.g.girard@free.fr>,
## updated for ACE 5.7.4 by Pau Garcia i Quiles <pgquiles@elpauer.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix failures and warnings with g++ 4.3. See 
## DP: http://bugzilla.dre.vanderbilt.edu/show_bug.cgi?id=3270

@DPATCH@

diff -rupd ACE_wrappers.orig/ace/config-macros.h ACE_wrappers/ace/config-macros.h
--- ACE_wrappers.orig/ace/config-macros.h	2009-11-23 01:19:57.000000000 +0100
+++ ACE_wrappers/ace/config-macros.h	2009-11-24 00:01:54.000000000 +0100
@@ -385,7 +385,7 @@
 // This eliminates the warnings, and no code is generated for the null
 // conditional statement.  @note that may only be true if -O is enabled,
 // such as with GreenHills (ghs) 1.8.8.
-#  define ACE_UNUSED_ARG(a) do {/* null */} while (&a == 0)
+#  define ACE_UNUSED_ARG(a) do {/* null */} while (&a < 0)
 # elif defined (__DMC__)
    #define ACE_UNUSED_ID(identifier)
    template <class T>
